flang/docs/ParallelFortranRuntime.md

```diff
@@ -0,0 +1,1987 @@
+  referenced variable shall be of type `prif_notify_type`, and the storage
+  must have been allocated using `prif_allocate_coarray` or `prif_allocate`.
+  If this optional argument is omitted, then no notification is performed.
+
+* **`remote_ptr`**: pointer to where on the identified image the data begins.
+  The referenced storage must have been allocated using `prif_allocate_coarray` or `prif_allocate`.
+
+### `prif_put`
+
+**Description**: This procedure assigns to the elements of a coarray, when the elements to be
+  assigned are contiguous in linear memory on both sides. 
+  The compiler can use this to implement assignment to a _coindexed-named-object_. 
+  It need not call this procedure when the coarray reference is not a _coindexed-named-object_. 
+  This procedure blocks on source completion. This procedure corresponds to a contiguous coarray
+  reference on the left hand side of an _assignment-stmt_.
+    
+```
+subroutine prif_put( &
+    coarray_handle, cosubscripts, value, first_element_addr, &
+    team, team_number, notify_ptr, stat, errmsg, errmsg_alloc)
+  type(prif_coarray_handle), intent(in) :: coarray_handle
+  integer(c_intmax_t), intent(in) :: cosubscripts(:)
+  type(*), intent(in), contiguous, target :: value(..)
+  type(c_ptr), intent(in) :: first_element_addr
+  type(prif_team_type), optional, intent(in) :: team
+  integer(c_intmax_t), optional, intent(in) :: team_number
+  integer(c_intptr_t), optional, intent(in) :: notify_ptr
+  integer(c_int), intent(out), optional :: stat
+  character(len=*), intent(inout), optional :: errmsg
+  character(len=:), intent(inout), allocatable, optional :: errmsg_alloc
+end subroutine
+```
+**Further argument descriptions**:
+
+* **`first_element_addr`**: The address of the current image's data in the coarray
+  corresponding to the first element to be assigned to on the identified image
+
+### `prif_put_raw`
+
+**Description**: Assign to `size` number of contiguous bytes on given image, starting at
+  `remote_ptr`, copying from `current_image_buffer`.
+    
+```
+subroutine prif_put_raw( &
+    image_num, current_image_buffer, remote_ptr, notify_ptr, size, &
+    stat, errmsg, errmsg_alloc)
+  integer(c_int), intent(in) :: image_num
+  type(c_ptr), intent(in) :: current_image_buffer
+  integer(c_intptr_t), intent(in) :: remote_ptr
+  integer(c_intptr_t), optional, intent(in) :: notify_ptr
+  integer(c_size_t), intent(in) :: size
+  integer(c_int), intent(out), optional :: stat
+  character(len=*), intent(inout), optional :: errmsg
+  character(len=:), intent(inout), allocatable, optional :: errmsg_alloc
+end subroutine
+```
+**Further argument descriptions**:
+
+* **`image_num`**: identifies the image to be written to in the initial team
+* **`current_image_buffer`**: pointer to the contiguous data which should be copied to the
+  identified image. 
+* **`size`**: how much data is to be transferred in bytes
+
+### `prif_put_raw_strided`
```

Comment:
Quoted from @jeffhammond's response:

> `prif_{get,put}_strided` need to exist to allow for an efficient implementation over MPI. The MPI_Win argument is going to be in type(`prif_coarray_handle`) and therefore that argument needs to be present for all communication that is going to lead to MPI RMA.

Reply:
**LBL response:** We will add `prif_{get,put}_strided` to the upcoming PRIF version.